### ISSUE TITLE
Unable to resize with mouse, right half of window not clickable Ubuntu 24.04

### ISSUE THREAD

**Issue by liquidcarbon** [microsoft/vscode#210624]
Same symptoms as in #188407 that go away with `code --ozone-platform=wayland`

>  any attempt to resize the application window larger, or click maximize, results in the application appearing to be fullscreen, but any space larger than the original window the app opened in is unclickable. If another window is behind the app in that space, it will click whatever is behind it. Tried this using VS Code from the Ubuntu Software app (snap), and same result using the .deb binary downloaded directly from Microsoft. No extensions installed, happens with fresh install.

Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.88.1
- OS Version: Ubuntu 24.04

Steps to Reproduce:

not sure

**Comment by qnixdev** [comment]
Similar symptoms with windows in IntelliJ Idea all versions. After any submenu in the window is minimized, the window cannot be resized. When you move the cursor to the edge of the menu window, its size does not change.
![Screenshot from 2024-04-28 17-53-23](https://github.com/microsoft/vscode/assets/73113274/0fd60196-6ce7-459a-86c1-3a1b15984808)

**Comment by leixiao-nv** [comment]
Log out then log in, the problem wih snap version of vscode 1.88.1 is disapeared

**Comment by liheyuan** [comment]
may be a wayland problem see https://github.com/microsoft/vscode/issues/212430

**Comment by RamonPessoa** [comment]
Same problem here and i'm not using snap version, after logout and login it works fine. I'm not sure if it helps, but  I use dual monitor.

**Comment by JashKakadiya** [comment]
yes it is work for me (ubuntu 24.04) , just logout and re login

**Comment by aldolammel** [comment]
> Same symptoms as in #188407 that go away with `code --ozone-platform=wayland`
> 
> > any attempt to resize the application window larger, or click maximize, results in the application appearing to be fullscreen, but any space larger than the original window the app opened in is unclickable. If another window is behind the app in that space, it will click whatever is behind it. Tried this using VS Code from the Ubuntu Software app (snap), and same result using the .deb binary downloaded directly from Microsoft. No extensions installed, happens with fresh install.
> 
> Does this issue occur when all extensions are disabled?: Yes/No
> 
> * VS Code Version: 1.88.1
> * OS Version: Ubuntu 24.04
> 
> Steps to Reproduce:
> 
> not sure

Same issue for me. I'm working in a Ubuntu laptop. Last version of everything in here. If I maximize the VSCode, the right side of VScode window stop to be clickable. I'm trying to code with PyCharm community here and everything is fine. So, yeap, it's a VSCode stuff.

**Comment by JashKakadiya** [comment]
> > Same symptoms as in #188407 that go away with `code --ozone-platform=wayland`
> > > any attempt to resize the application window larger, or click maximize, results in the application appearing to be fullscreen, but any space larger than the original window the app opened in is unclickable. If another window is behind the app in that space, it will click whatever is behind it. Tried this using VS Code from the Ubuntu Software app (snap), and same result using the .deb binary downloaded directly from Microsoft. No extensions installed, happens with fresh install.
> > 
> > 
> > Does this issue occur when all extensions are disabled?: Yes/No
> > 
> > * VS Code Version: 1.88.1
> > * OS Version: Ubuntu 24.04
> > 
> > Steps to Reproduce:
> > not sure
> 
> Same issue for me. I'm working in a Ubuntu laptop. Last version of everything in here. If I maximize the VSCode, the right side of VScode window stop to be clickable. I'm trying to code with PyCharm community here and everything is fine. So, yeap, it's a VSCode stuff.
> 
> Solution: I solved the issue, uninstalling my VSCode, going to Microsoft VScode website, downloading the last version again, installing via local file (not snap) and done. VSCode found my credential and stuff. Everything working fine as gold again.

Have you tried  logout and re login of your profile ? it is worked for me

**Comment by liheyuan** [comment]
Try Change vscode configuration, Window: Title Bar Style to Custom (instead of Native)

**Comment by aldolammel** [comment]
> Solution: I solved the issue, uninstalling my VSCode, going to Microsoft VScode website, downloading the last version again, installing via local file (not snap) and done. VSCode found my credential and stuff. Everything working fine as gold again.

No, the issue came back again. And it restarted when I opened two VSCode windows. Two hypotheses:
1) It starts when two or more VSCode windows are opened in the same session;
2) or when I transfer one of the windows to another desktop in my Ubuntu (Ubuntu 24.04 LTS).

**Comment by aldolammel** [comment]
> Try Change vscode configuration, Window: Title Bar Style to Custom (instead of Native)

It didn't work.

**Comment by aldolammel** [comment]
Interesting what this user figured out: 
https://github.com/microsoft/vscode/issues/212430#issue-2289453878

If you turn the Wayland (GPU Acceleration method) in your Ubuntu, VSCode will work properly again.

1) Do this:
$ sudo nano /etc/gdm3/custom.conf

2) Uncomment this line:
WaylandEnable=false

3) Restart your system. Enjoy.

More about:
https://jumpcloud.com/support/troubleshooting-remote-assist-disable-wayland-or-gpu-rendering-for-linux#:~:text=WaylandEnable%3Dfalse%20forces%20your%20system,%3Dfalse%20to%20WaylandEnable%3Dtrue%20.

**Comment by vlopatko** [comment]
I have the same problem with VScode(only with these application). (Ubuntu 24.04 LTS, VMware Player 17)
When I try to resize/expand the window, the right half of the VScode window remains inactive, and the ability to stretch the window size also disappears (the slider in the form of two arrows does not appear). Any right-click on the right half of the expanded VSCode window is perceived as a click on the desktop. 
![photo_2024-07-06_06-57-22](https://github.com/microsoft/vscode/assets/101696452/0a21127f-e8a6-4ee6-a018-a0253b8ff738)
It turned out that the problem is in the "Default keyring". This window appears when VScode starts, after entering the password, it disappears and all the above problems begin. To solve the problem, I had to go to settings -> "Passwords and keys" and remove the password for the "Default keyring". After these manipulations, the VScode window began to work as before.

**Comment by Quraian** [comment]
> Try Change vscode configuration, Window: Title Bar Style to Custom (instead of Native)

This is what worked for me.

**Comment by 0815fips** [comment]
Definitely a wayland problem. I had this issue with earlier Ubuntu and VSCodium versions. Disabling Wayland doesn't work for me on Ubuntu 24.04, because it simply won't boot.

**Comment by wesley990** [comment]
> Try Changing vs code configuration, Window: Title Bar Style to Custom (instead of Native)



> Try Changing vs code configuration, Window: Title Bar Style to Custom (instead of Native)

I am using VS code (installed via Snap) on Untubu 24.04. Setting the Title Bar Style from 'Native' to 'Custom' solved this issue.

**Comment by Mcklmo** [comment]
I have this issue not only with VS Code, but also with Chromium and Postman.

For VS Code and Chromium, i can get it to work by executing the applications with `--enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform-hint=auto
`.

May be unrelated, but Postman responds with the following error when I try this solution.

```bash
$ postman --enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform-hint=auto
The disableGPU setting is set to undefined
Not disabling GPU
[76653:0728/145801.592046:ERROR:wayland_connection.cc(214)] Failed to connect to Wayland display
[76653:0728/145801.592081:ERROR:ozone_platform_wayland.cc(227)] Failed to initialize Wayland platform
[76653:0728/145801.592085:ERROR:env.cc(226)] The platform failed to initialize.  Exiting.
The futex facility returned an unexpected error code.Aborted (core dumped)
```

**Comment by liheyuan** [comment]
Try upgrade to latest Ubuntu 24.04.1

It seems mutter ralated pkg fix this problem

mutter (46.2-1ubuntu0.24.04.1) noble; urgency=medium

[ Daniel van Vugt ]

New upstream release 46.2 (LP: #2068598)
Fixed mouse clicks falling through to the window behind (LP: #2026194)
...

**Comment by notPlancha** [comment]
> Try Change vscode configuration, Window: Title Bar Style to Custom (instead of Native)

this fixed it for me and I'm using x11

**Comment by rbudnar** [comment]
I ran into this issue as well in ubuntu 24.04 after upgrading from 22.04. The solution I found [here](https://www.reddit.com/r/Ubuntu/comments/1cdh92x/wayland_resizing_windows_problembug/) was to disable auto-login, which worked.

**Comment by woodytang** [comment]
> Try Change vscode configuration, Window: Title Bar Style to Custom (instead of Native)

 this works on VS code ,but not on cursor ~~

**Comment by woodytang** [comment]
I saw a user  on reddit  named SkyEqual2345 who had given a quick solution:

Workaround - super key+middle click on the title bar, then drag.

https://bugs.launchpad.net/ubuntu/+source/gtk+3.0/+bug/2064177

this works perfect for me, don't waste your time on a bug!
